src/components/BloombergToFDC3/components/RuleForm.tsx

```diff
@@ -0,0 +1,151 @@
+          data: "bbg.security",
+          type: "BloombergCommand",
+          args: {
+            tails,
+            panel
+          }
+        }
+      };
+
+      resetForm();
+
+      const response = await FSBL.Clients.ConfigClient.get(LINK_PREFERENCES_PATH);
+
+      if (!response.err) {
+        let links = response.data;
+        if (!Array.isArray(links)) {
+          links = [];
+        }
+        links[activeLink?.index ?? links.length] = value;
+
+        await FSBL.Clients.ConfigClient.setPreference({
+          field: LINK_PREFERENCES_PATH.join("."),
+          value: links,
+        });
+      }
+    } else {
+      setErrorString("You must set a terminal command");
+    }
+  }
+
+  return <div>
+    <div className="error-warning" aria-disabled={errorString == ""}>{errorString}</div>
+    <table role="presentation" className="full-width">
+      <tbody>
+        <tr>
+          <th>Command</th>
+          <td>
+            <input value={command} aria-label="Command" type="text" size={5} maxLength={9} placeholder="DES"
+              className="mnemonic-input"
+              onChange={(e) => {
+                setCommand(e.target.value)
+            }} />
+            <input value={tails} aria-label="Command arguments (Tails)" type="text" size={5} maxLength={64} placeholder="args"
+              className="stretchy-input"
+              onChange={(e) => {
+                setTails(e.target.value)
+            }} />
+          </td>
+        </tr>
+        <tr>
+          <th>Panel</th>
+          <td>  
+            <select value={panel} onChange={(e) => setPanel(e.target.value)}>
+              <option value={0}>0</option>
+              <option value={1}>1</option>
+              <option value={2}>2</option>
+              <option value={3}>3</option>
+              <option value={4}>New tab</option>
+            </select>
+          </td>
+        </tr>
+        <tr>
+          <th>FDC3 Intent</th>
+          <td><select value={intent} className="stretchy-input"
+              onChange={(e) => setIntent(e.target.value)}>
+            <option value="">Select Intent</option>
+            <option value="CreateTradeTicket">CreateTradeTicket</option>
+            <option value="ViewChart">ViewChart</option>
+            <option value="ViewAnalysis">ViewAnalysis</option>
+            <option value="ViewHoldings">ViewHoldings</option>
+            <option value="ViewInstrument">ViewInstrument</option>
+            <option value="ViewNews">ViewNews</option>
+            <option value="ViewOptions">ViewOptions</option>
+            <option value="ViewOrders">ViewOrders</option>
+            <option value="ViewQuote">ViewQuote</option>
+            <option value="ViewResearch">ViewResearch</option>
+          </select>
+          </td>
+        </tr>
+        <tr>
+          <th>Display Name</th>
+          <td>
+            <input type="text" value={displayName ?? ""} size={5} maxLength={64} placeholder="optional"

```

Comment:
There shouldn't actually be a point where `displayName` is undefined, so the `?? ""` shouldn't ever be needed. I'm surprised typescript didn't complain about unreachable code.

Reply:
thanks, fixed

Reply:
(having said that... not sure if it will make any difference if it is undefined)